src/Mvc/Mvc.Abstractions/src/ModelBinding/Metadata/ModelMetadataIdentity.cs

```diff
@@ -47,6 +49,7 @@ public static ModelMetadataIdentity ForType(Type modelType)
         /// <param name="name">The name of the property.</param>
         /// <param name="containerType">The container type of the model property.</param>
         /// <returns>A <see cref="ModelMetadataIdentity"/>.</returns>
+        [Obsolete("This API is obsolete and may be removed in a future release.")]
```

Comment:
Breaking change? In 3.1? Is this something we simply forgot to do earlier?
We may need to avoid doing this.

Reply:
Actually, I now see why you're obsoleting this, because we will be recommending customers to use the newer factory method.

Reply:
Thinking more about this, can we move this behind a flag? We can't simply make a breaking change without tactics, and I believe tactics won't approve this.

Reply:
Adding an Obsolete attribute isn't a breaking change. It's binary compatible, and you may suppress the warning and continue using it.